src/app/footer/footer.ts

```diff
@@ -0,0 +1,56 @@
+import {Component} from 'angular2/core';
+import {Location} from 'angular2/router';
+
+@Component({
+  selector: 'publish-footer',
+  styleUrls: ['app/footer/footer.css'],
+  template: `
```

Comment:
meant it as an honest question. I agree on the highlighting - seems like from an editing standpoint, that is a trade-off, but it is nicer to have inline where the inputs, directives, and whatnot are also right in front of you.